WordPressKit/PostServiceRemoteREST+Extended.swift

```diff
@@ -40,7 +40,7 @@ private func decodePost(from object: AnyObject) async throws -> RemotePost {
 
 private func makeParameters<T: Encodable>(from value: T) throws -> [String: AnyObject] {
     let encoder = JSONEncoder()
-    encoder.dateEncodingStrategy = .formatted(NSDate.rfc3339DateFormatter())
+    encoder.dateEncodingStrategy = .formatted(NSDate.rfc3339DateFormatter)
```

Comment:
I forgot about the API breakage concerns when making this a `static let`. 🤦‍♂️ 

Turns out there is one use of this outside the library, [in the WordPress iOS tests](https://github.com/search?q=repo%3Awordpress-mobile%2FWordPress-iOS+rfc3339DateFormatter&type=code). 

I'll update the code reintroducing the `static func` but marking it as deprecated. This way, we can remove it at the next "serious" breaking change occasion.